apps/server/src/modules/saga/sagas/user-deletion.ts

```diff
@@ -0,0 +1,69 @@
+import { Saga } from '../interface/saga';
+import { SagaStep } from '../interface/saga-step';
+import { SagaInjectionService } from '../service/injection';
+import { SagaCompensateFn } from '../type/saga-compensate-function';
+import { DeletionRequest } from '../../deletion/domain/do';
+
+type Compensation = { fn: SagaCompensateFn; step: SagaStep };
+
+export class UserDeletionSaga implements Saga {
+	name = 'user-deletion';
+
+	// in case we want to allow parallel execution of steps we could have a list of lists
+	flow: SagaStep['sagaName'][] = [
+		'delete-user-from-class',
+		'delete-user-from-news', // not implemented yet
+	];
```

Comment:
I would define the steps in deletion module itself and call invoke with the steps
it is there that we have the domains, and any of the business logic